src/Integrations/Integrations/DatabaseIntegrationHelper.php

```diff
@@ -43,7 +43,11 @@ public static function injectDatabaseIntegrationData(HookData $hook, $backend, $
                 $propagationMode = \DDTrace\DBM_PROPAGATION_SERVICE;
             }
 
-            $query = self::propagateViaSqlComments($hook->args[$argNum], $hook->span()->service, $propagationMode);
+            $databaseService = isset($hook->span()->meta['peer.service'])
+                ? $hook->span()->meta['peer.service']
+                : $hook->span()->service;
```

Comment:
I may push my nit a little bit too far, but maybe we could save some `hook->span()` calls with something along these lines considering we create a span anyway? Considering the staggering amount of calls that are made on these SQL functions, it may be worth 🤷 
```php
$span = $hook->span();
$databaseService = isset($span->meta['peer.service'])
                ? $span->meta['peer.service']
                : $span->service;
```

Reply:
Also use `??` instead.
```suggestion
            $span = $hook->span();
            $databaseService = $span->meta['peer.service'] ?? $span->service;
```